src/mono/mono/metadata/icall.c

```diff
@@ -4246,6 +4246,17 @@ ves_icall_RuntimeType_GetPropertiesByName_native (MonoQCallTypeHandle type_handl
 	return NULL;
 }
 
+gpointer
+ves_icall_RuntimeType_GetValueConstructor (MonoObjectHandleOnStack ref_ctor_info, MonoError *error)
+{
+	gpointer ctor_fptr;
+	MonoReflectionMethod* ctor = *(MonoReflectionMethod**)ref_ctor_info;
+	ctor_fptr = mono_compile_method_checked (ctor->method, error);
+	if (!is_ok (error))
+		return NULL;
```

Comment:
this will cause a managed exception to be thrown if there's an `error` (for example if we trigger assembly loading and the assembly isn't found, or if the method uses an invalid class we'll throw `BadImageFormatException` or something)  is that what you want? or do you just want to return null and throw a generic `ExecutionEngineException` up in managed?

Reply:
I want the exception to be thrown if it exists. If an exception isn't thrown, but `NULL` is returned, we will throw an `ExecutionEngineException`.